scripts/src/java/org/oppia/android/scripts/maven/GenerateMavenDependenciesList.kt

```diff
@@ -0,0 +1,546 @@
+  }
+
+  val finalLicensesSet = updateLicensesSet(
+    licenseSetFromTextproto,
+    licenseSetFromTextproto
+  )
+
+  val finalDependenciesList = updateMavenDependenciesList(
+    dependenciesListFromPom,
+    finalLicensesSet
+  )
+  writeTextProto(
+    pathToMavenDependenciesTextProto,
+    MavenDependencyList.newBuilder().addAllMavenDependency(finalDependenciesList).build()
+  )
+
+  val licensesToBeFixed = getAllBrokenLicenses(finalDependenciesList)
+  if (licensesToBeFixed.isNotEmpty()) {
+    println("Please provide all the details of the following licenses manually:")
+    licensesToBeFixed.forEach {
+      println("\nlicense_name: ${it.licenseName}")
+      println("primary_link: ${it.primaryLink}")
+      println("primary_link_type: ${it.primaryLinkType}")
+      println("secondary_link: ${it.secondaryLink}")
+      println("secondary_link_type: ${it.secondaryLinkType}")
+      println("secondary_license_name: ${it.secondaryLicenseName}\n")
+    }
+    throw Exception("Licenses details are not completed.")
+  }
+
+  val dependenciesWithoutAnyLinks = getDependenciesWithNoAndInvalidLinks(finalDependenciesList)
+  if (dependenciesWithoutAnyLinks.isNotEmpty()) {
+    println("Please provide the license links for the following dependencies manually:")
+    dependenciesWithoutAnyLinks.forEach {
+      println(it)
+    }
+    throw Exception(
+      """
+      There does not exist any license links (or the extracted license links are invalid) 
+      for some dependencies.
+      """.trimIndent()
+    )
+  }
+
+  val dependenciesThatNeedHumanIntervention = getDependenciesThatNeedHumanIntervention(
+    finalDependenciesList
+  )
+  if (dependenciesThatNeedHumanIntervention.isNotEmpty()) {
+    println(
+      """There are still some dependencies that need human intervention.
+      Try to find the license links for these dependencies and coordinate with the Oppia-android maintainers
+      to fix the issue.
+      """.trimIndent()
+    )
+    throw Exception("Human Intervention needed.")
+  }
+  println("Maven Dependencies updated successfully.")
+}
+
+private fun updateLicensesSet(
+  licenseSetFromTextproto: Set<License>,
+  licenseSetFromPom: Set<License>
+): Set<License> {
+  val finalLicensesSet = mutableSetOf<License>()
+  licenseSetFromPom.forEach { license ->
+    val updatedLicense = licenseSetFromTextproto.find { it.primaryLink == license.primaryLink }
+    if (updatedLicense != null) {
+      finalLicensesSet.add(updatedLicense)
+    } else {
+      finalLicensesSet.add(license)
+    }
+  }
+  return finalLicensesSet
+}
+
+private fun getDependenciesThatNeedHumanIntervention(
+  mavenDependenciesList: List<MavenDependency>
+): Set<MavenDependency> {
+  val dependencies = mutableSetOf<MavenDependency>()
+  mavenDependenciesList.forEach { dependency ->
+    dependency.licenseList.forEach { license ->
+      if (license.primaryLinkType == PrimaryLinkType.NEEDS_INTERVENTION) {
+        dependencies.add(dependency)
+      }
+    }
```

Comment:
Extending from my Kotlin idiomatic comment above, this could be replaced with a filter{} I think. Ditto for other similar cases.

Reply:
Done.